tests/proxy_test_helper.c

```diff
@@ -265,10 +265,10 @@ int proxy_tester_create_testing_channel_connection(struct proxy_tester *tester)
     ASSERT_NOT_NULL(slot);
     ASSERT_SUCCESS(aws_channel_slot_insert_end(tester->testing_channel->channel, slot));
     ASSERT_SUCCESS(aws_channel_slot_set_handler(slot, &connection->channel_handler));
-    connection->vtable->on_channel_handler_installed(&connection->channel_handler, slot);
+    testing_channel_drain_queued_tasks(tester->testing_channel);
 
+    connection->vtable->on_channel_handler_installed(&connection->channel_handler, slot);
     tester->client_connection = connection;
-    testing_channel_drain_queued_tasks(tester->testing_channel);
```

Comment:
Why did we need to move `testing_channel_drain_queued_tasks()` between installing the handler and invoking its `on_channel_handler_installed()` callback?

If you needed to drain the tasks exactly here in tests, its not going to work in the real world. The actual function that installs the http-channel-handler outside of tests invokes the callback right away